### PR TITLE
DPR2-128: Skip failed maintenance and error only at the end.

### PR DESCRIPTION
**Pull request by tom-ogle-moj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/service/MaintenanceService.java [hunk 1]
```diff
@@ -24,26 +27,52 @@ public MaintenanceService(DataStorageService storageService) {
...
      */
-    public void compactDeltaTables(SparkSession spark, String rootPath) throws DataStorageException {
+    public void compactDeltaTables(SparkSession spark, String rootPath) throws DataStorageException, MaintenanceOperationFailedException {
         logger.info("Beginning delta table compaction for tables under root path: {}", rootPath);
         List<String> deltaTablePaths = storageService.listDeltaTablePaths(spark, rootPath);
-        logger.info("Found delta tables at: {}", String.join(", ", (deltaTablePaths)));
-        for (String deltaTablePath : deltaTablePaths) {
-            storageService.compactDeltaTable(spark, deltaTablePath);
-        }
+        logger.info("Found delta tables at: {}", java.lang.String.join(", ", (deltaTablePaths)));
```
Comment (gavriil-g-moj) [inline src/main/java/uk/gov/justice/digital/service/MaintenanceService.java:33]: This was already there so not a new change but does this cause too much logging or are the delta table paths not so many to flood the logs?

**Review by gavriil-g-moj — APPROVED** [review]
LGTM

**Review by koladeadewuyi-moj — APPROVED** [review]

File: src/main/java/uk/gov/justice/digital/service/MaintenanceService.java (hunk 1 above)
Comment (tom-ogle-moj) [reply src/main/java/uk/gov/justice/digital/service/MaintenanceService.java:33]: Good point. It is 6 right now but could be a lot more in the future, in which case the string join would be nasty. I'm not sure why intellij didn't fix the java.lang.String import either. 

How about I change it to log the number of tables found instead for both compact and vacuum?

Comment (gavriil-g-moj) [reply src/main/java/uk/gov/justice/digital/service/MaintenanceService.java:33]: Changing it to log the number of tables sounds like a good idea. 
If the actual paths are needed for debugging purposes maybe this log can be changed to debug level.

Comment (tom-ogle-moj) [reply src/main/java/uk/gov/justice/digital/service/MaintenanceService.java]: Have done so now

**Review by gavriil-g-moj — APPROVED** [review]

**Pull request merged**